packages/commonwealth/client/scripts/views/pages/manage_community/ManageCommunityPage.tsx

```diff
@@ -57,48 +70,51 @@ const ManageCommunityPage = () => {
     setMods(memberMods);
   };
 
-  const searchMembers = async (searchQuery?: string) => {
-    try {
-      let profiles = [];
-
-      const cachedResult = membersCache.get(searchQuery);
-      if (cachedResult) {
-        profiles = cachedResult.profiles;
-      } else {
-        const res = await axios.get(`${app.serverUrl()}/searchProfiles`, {
-          params: {
-            chain: app.activeChainId(),
-            search: searchQuery || '',
-            page_size: 100,
-            page: 1,
-            include_roles: true,
-          },
-        });
-        if (res.data.status !== 'Success') {
-          throw new Error('Could not fetch members');
-        }
-        membersCache.set(searchQuery, res.data.result);
-        profiles = res.data.result.profiles;
-      }
-
-      let roles = [];
+  const fetchSearchResults = async ({ pageParam = 0 }) => {
+    const {
+      data: { result },
+    } = await axios.get<{ result: ProfilesSearchResponse }>(`/api/profiles`, {
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      params: {
+        chain: app.activeChainId(),
+        search: searchTerm,
+        limit: (50).toString(),
+        page: pageParam.toString(),
+        order_by: orderBy,
+        order_direction: orderDirection,
+        include_roles: 'true',
+      },
+    });
+    return result.results;
+  };
 
-      if (profiles.length > 0) {
-        roles = profiles.map((profile) => {
-          return {
-            ...(profile.roles[0] || {}),
-            Address: profile.addresses[0],
-            id: profile.addresses[0].id,
-          };
-        });
-      }
-      setRoleData(roles);
-      setInitialized(true);
-    } catch (err) {
-      setRoleData([]);
-      setInitialized(true);
+  const { data: searchResults, refetch } = useQuery({
+    queryKey: [
+      'search-members-manage',
```

Comment:
we try to avoid string as a key. Take a look on other react-query examples in the code like in `state/api/comments/fetchReactions` eg
`queryKey: [ApiEndpoints.getCommentReactions(commentId), chainId],`